src/main.rs

```diff
@@ -51,12 +55,14 @@ fn main() -> iced::Result {
         },
     };
 
-    let config = toml::from_str::<Config>(&config_file).unwrap_or_else(|e| {
+    let mut config = toml::from_str::<Config>(&config_file).unwrap_or_else(|e| {
         tracing::error!("failed to parse config: {e:#?}");
         let config = Config::default();
         tracing::warn!("falling back to default configs: {config:#?}");
         config
     });
+    config.url = url;
```

Comment:
the url will be parsed as `None` when changed to `Option<String>`

```suggestion
    let config = toml::from_str::<Config>(&config_file).unwrap_or_else(|e| {
        tracing::error!("failed to parse config: {e:#?}");
        let config = Config::default();
        tracing::warn!("falling back to default configs: {config:#?}");
        config
    });
```